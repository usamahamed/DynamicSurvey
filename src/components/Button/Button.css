.button {
  display: inline-block;
  height: 40px;
  padding: 0 14px;
  border: 0;
  border-radius: 4px;
  outline: 0;
  background-color: #fff;
  box-shadow: 0 4px 6px rgba(50, 50, 93, 0.11), 0 1px 3px rgba(0, 0, 0, 0.08);
  font-size: 15px;
  font-weight: 500;
  line-height: 40px;
  text-decoration: none;
  cursor: pointer;
  transition: all 0.15s ease;
}

.button:hover {
  box-shadow: 0 7px 14px rgba(50, 50, 93, 0.3), 0 3px 6px rgba(0, 0, 0, 0.08);
}

.button--large {
  height: 50px;
  padding: 5px 15px;
}

.button--flat {
  background-color: #f9f9f9;
  box-shadow: none;
}

.button--flat:hover {
  box-shadow: none;
}

.button--success {
  background-color: #00a562;
  color: #fff;
}

.button:disabled {
  background-color: #024e2f;
}
