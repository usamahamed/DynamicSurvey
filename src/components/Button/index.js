import React from 'react';
import PropTypes from 'prop-types';

import './Button.css';

const Button = props => (
  <button
    className={`button ${props.className}`}
    disabled={props.fieldValue === ''}
    onClick={props.buttonClick}
  >
    {props.children}
  </button>
);

Button.defaultProps = {
  className: '',
  fieldValue: null,
};

Button.propTypes = {
  buttonClick: PropTypes.func.isRequired,
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.element])
    .isRequired,
  className: PropTypes.string,
  fieldValue: PropTypes.string,
};

export default Button;
