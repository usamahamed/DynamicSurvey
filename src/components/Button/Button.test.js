import React from 'react';
import { shallow } from 'enzyme';
import Button from './index';

it('renders the button', () => {
  const onButtonClick = jest.fn();
  const wrapper = shallow(<Button buttonClick={onButtonClick}>HELLO</Button>);
  expect(wrapper).toMatchSnapshot();
});

it('responds to click', () => {
  const onButtonClick = jest.fn();
  const wrapper = shallow((<Button buttonClick={onButtonClick}>TEXT</Button>));
  wrapper.find('button').simulate('click');
  expect(wrapper).toMatchSnapshot();
});
