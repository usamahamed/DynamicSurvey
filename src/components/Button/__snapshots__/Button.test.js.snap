// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the button 1`] = `
<button
  className="button "
  disabled={false}
  onClick={[Function]}
>
  HELLO
</button>
`;

exports[`responds to click 1`] = `
<button
  className="button "
  disabled={false}
  onClick={[Function]}
>
  TEXT
</button>
`;
